vscode-extension/src/features/diagnostics/Diagnostics.ts

```diff
@@ -58,29 +60,30 @@ export class Diagnostics implements Disposable {
     //#endregion
 
     private readonly _diagnostics: DiagnosticCollection;
-    private _documents = new WeakMap<Uri, PropertyBag<Diagnostic[]>>();
-    private _schedulers = new WeakMap<Uri, NodeJS.Timeout>();
+    private _documents = new Map<string, PropertyBag<Diagnostic[]>>();
```

Comment:
Why `Map` instead of `WeakMap`? You depend on `fsPath` as key, whereas in the past we could use any URI.

Is there something that didn't work with `WeakMap` and URIs as keys? If so, then fine. If not, then please use `WeakMap`.